commit 403fbcb56626551fe911af0c3a5ee2b5122df664
Author: Sergei Tachenov <sergei@tachenov.name>
Date:   Thu Jun 23 21:24:23 2022 +0300

    Use C++17 for Qt 6
    
    Apparently Qt 6 requires C++17 (see #145). But we don't want to
    restrict everyone to C++17 just because Qt 6 requires it. So we
    set it based on the selected (or auto-selected) Qt version.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a3eaf1a..6c7c6c7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,8 +3,6 @@ cmake_minimum_required(VERSION 3.15...3.18)
 
 project(QuaZip VERSION 1.3)
 
-set(CMAKE_CXX_STANDARD 14)
-
 set(QUAZIP_LIB_VERSION ${QuaZip_VERSION})
 set(QUAZIP_LIB_SOVERSION 1.3.0)
 
@@ -34,6 +32,12 @@ endif()
 
 set(QUAZIP_QT_MAJOR_VERSION ${QT_VERSION_MAJOR} CACHE STRING "Qt version to use (4, 5 or 6), defaults to ${QT_VERSION_MAJOR}")
 
+if (QUAZIP_QT_MAJOR_VERSION EQUAL 6)
+    set(CMAKE_CXX_STANDARD 17)
+else()
+    set(CMAKE_CXX_STANDARD 14)
+endif()
+
 if(NOT CMAKE_BUILD_TYPE)
 	set(CMAKE_BUILD_TYPE RELEASE)
 endif()
